### PR TITLE
Debian package support

### PR DESCRIPTION
**Pull request by derfel**
This PR enable the creation of debian packages from srt source code:
- srt-bin: stransmit and sfplay
- libsrt1: libsrt
- libsrt-dev: dev files for libsrt

### CODE REVIEW

**Comment by tdprime** [comment]
What is the status of this?

I too created a branch with packaging metadata appropriate for Ubuntu 16.04.
https://github.com/tdprime/srt/tree/debian

**Comment by maxlovic** [comment]
Hi @derfel 
Can you please update this PR?

Or maybe @tdprime can create a new one?

**Comment by dabrain34** [comment]
Calling `dpkg-buildpackage -us -uc` fails with error on a fresh repository:

```
dh_install: libsrt-dev missing files: usr/lib/*/lib*.so
dh_install: libsrt-dev missing files: usr/lib/*/pkgconfig/*
dh_install: libsrt1 missing files: usr/lib/*/lib*.so.*
dh_install: missing files, aborting
debian/rules:4: recipe for target 'binary' failed
make: *** [binary] Error 255
```

I would say that the options should be also important such as gnutls

**Comment by dabrain34** [comment]
why is it ``abandonned`` ?

**Comment by maxlovic** [comment]
The original author is missing.

**Comment by dabrain34** [comment]
I can try to fix it ...

**Comment by ethouris** [comment]
We set it abandoned because the original author forgot it and we had no one else capable of working on it. I can restore it to some reasonable status if you can fix it.

**Comment by dabrain34** [comment]
ok I put it in my todo list then :)

**Comment by dabrain34** [comment]
After further investigation, I can approve this patch seen that I was not building with the proper steps.
I did not appy the patch ``00_cmake_multiarch`` from debian/patches with ``quilt``

Here is what I did to generate the deb package:

- install quilt
```
# sudo pat install quilt
```

- Read this [wiki](https://raphaelhertzog.com/2012/08/08/how-to-use-quilt-to-manage-patches-in-debian-packages/) about quilt in deb package
- Update ~/.quiltrc to be debian compatible
```
for where in ./ ../ ../../ ../../../ ../../../../ ../../../../../; do
    if [ -e ${where}debian/rules -a -d ${where}debian/patches ]; then
        export QUILT_PATCHES=debian/patches
        break
    fi
done
```
 - validate that quilt now is correctly discovering the debian patches.
```
# quilt series
00_cmake_multiarch
```
 - Apply the patch
```
# quilt push
# quilt applied
00_cmake_multiarch
```
 - Then build the debian package
```
# dpkg-buildpackage -us -uc
```

**Comment by dabrain34** [comment]
https://packages.ubuntu.com/disco/libsrt1

**Comment by maxlovic** [comment]
@dabrain34 Cool, thanks!

File: debian/changelog [hunk 1]
```diff
@@ -0,0 +1,5 @@
+srt (1.2.0-1) unstable; urgency=medium
```
Comment (maxlovic) [inline debian/changelog]: Will package only release builds. Starting from 1.3.3 and above.

File: debian/control [hunk 2]
```diff
@@ -0,0 +1,42 @@
+Source: srt
+Priority: optional
+Maintainer: derfel <code@derfel.net>
```
Comment (maxlovic) [inline debian/control]: Probably the maintainer should be changed.

File: debian/control [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+Multi-Arch: same
+Depends: libsrt1 (= ${binary:Version}), ${misc:Depends}
+Description: SRT - Secure, Reliable, Transport
+ Secure Reliable Transport (SRT) is a proprietary transport
+ technology that optimizes streaming performance across
+ unpredictable networks, such as the Internet.
+ .
+ This package contains development files for libsrt1
+
+Package: libsrt1
```
Comment (maxlovic) [inline debian/control]: libsrt1 -> libsrt?

File: debian/rules [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/make -f
+
+%:
+	dh $@
+
+override_dh_clean:
+	dh_clean
+	rm -f CMakeCache.txt Makefile cmake_install.cmake haisrt.pc install_manifest.txt srt.pc
+	rm -f libsrt.so.1.2.0 stransmit utility-test libsrt.so.1 libsrt.so
```
Comment (maxlovic) [inline debian/rules]: libsrt.so.1.3.3
`stransmit` is no longer present. Now there are `srt-live-transmit` and `srt-file-transmit`.

File: debian/control (hunk 3 above)
Comment (ocrete) [reply debian/control]: Debian policy requires that the package name match exactly the soname including the version:

Ref: https://www.debian.org/doc/debian-policy/ch-sharedlibs.html#run-time-shared-libraries

File: debian/rules (hunk 4 above)
Comment (ocrete) [reply debian/rules]: that `rm -f` feels wrong, doesn't the cmake file generate a working "make clean"? If not, it should be fixed to do so.

**Comment by FedericoCeratto** [comment]
FYI There is already an official Debian package and the configuration files are in https://salsa.debian.org/debian/libsrt/tree/master/debian

**Comment by maxlovic** [comment]
Thank you @FedericoCeratto. Then there is no point to keep this PR open.

**Pull request closed without merging**